paasta_tools/api/views/autoscaler.py

```diff
@@ -45,6 +51,35 @@ def update_autoscaler_count(request):
     service = request.swagger_data.get('service')
     instance = request.swagger_data.get('instance')
     desired_instances = request.swagger_data.get('json_body')['desired_instances']
-    set_instances_for_marathon_service(service=service, instance=instance, instance_count=desired_instances)
-    response_body = {'desired_instances': desired_instances}
-    return Response(json_body=response_body, status_code=202)
+
+    try:
+        service_config = load_marathon_service_config(
+            service=service,
+            instance=instance,
+            cluster=settings.cluster,
+            soa_dir=settings.soa_dir,
+            load_deployments=False,
+        )
+    except:
+        error_message = 'Unable to load service config for %s.%s' % (service, instance)
+        raise ApiFailure(error_message, 404)
+
+    max_instances = service_config.get_max_instances()
+    if max_instances is None:
+        error_message = 'Autoscaling is not enabled for %s.%s' % (service, instance)
+        raise ApiFailure(error_message, 404)
+
+    min_instances = service_config.get_min_instances()
+
+    if desired_instances > max_instances:
+        set_instances_for_marathon_service(service=service, instance=instance, instance_count=max_instances)
+        error_message = 'desired_instances is set to max_instances %d' % max_instances
```

Comment:
It would be nice if these all returned in the same format to make it easier to parse